node/src/components/small_network/message.rs

```diff
@@ -44,6 +44,27 @@ impl<P: Payload> Message<P> {
     }
 }
 
+/// A pair of secret keys used by consensus.
+pub(super) struct ConsensusKeyPair {
+    secret_key: SecretKey,
```

Comment:
IIRC, the intent was to never clone a `SecretKey` but use an `Arc`.

Reply:
It is not cloned, as the outer structure is providing the `Arc` here. The `struct` itself does not implement `Clone` as well :)

Reply:
But you do pass `SecretKey` to the constructor, not `Arc`. So it has to be moved into this struct – which means it's either forgotten on the caller end or it's cloned, isn't that right?

Reply:
That depends on the compiler. It is moved, but whether or not that copies it from stack to heap here is not for us to specify.

I did notice that there have been some changes to how this is handled in `dev`, let's re-examine it after I have finished rebasing.

Reply:
Done. So the question remains if we want to be strict and move all responsibility of creating a heap-only copy into the `Loadable` impl of `Arc<SecretKey>`.

If that's the case, then it would be correct for `ConsensusKeyPair` to hold an `Arc<SecretKey>`.

On the other hand, I think the preferable alternative would be to alter `SecretKey` itself to always use the heap and security features instead, as that would make it impossible to construct stack-allocated keys at all. Otherwise we are relying on convention instead of types ("always use an `Arc` around `SecretKey`").

Reply:
So we actually _do_ end up with multiple copies in memory here, because we call `load_keys` twice, right?
I don't think that's an issue, but if it is, `load_keys` itself could memoize it the first time, and then only clone the `Arc`.

Reply:
That's fair, but I imagine the logic would be somewhat complicated, as we would have to ensure not to load the same key twice. Is that worth the effort right now? I feel it'd be better spent in a follow-up item, implementing a secure conclave for the key?

Reply:
I agree, I think it's fine as it is. :+1: